handlers/backend_handler.go

```diff
@@ -16,9 +16,20 @@ import (
 
 var TLSSkipVerify bool
 
-func NewBackendHandler(backendURL *url.URL, connectTimeout, headerTimeout time.Duration, logger logger.Logger) http.Handler {
+func NewBackendHandler(
+	backendID string,
```

Comment:
I'm not sure about the value of a `backendID` if backend URLs are expected to be unique. Could we use the URL as the ID here?

Reply:
I wasn't sure if using the URL or the ID was better, and didn't check in the database how they looked.

One side-effect of the backend ID, is that they are easier to type:

```
router_backend_handler_response_duration_seconds_count{backend_id="search-api"}
```

Also another "feature" is that if the backend URL is changed in the database, the metric will stay the same. Whether in practice this would ever happen is not something I know.

Presumably the IDs are more consistent across environments (staging/prod) than URLs? This is a question to which I do not know the answer.

What do you think is more ergonomic/intuitive?